open-notes/maturity-model.md

```diff
@@ -0,0 +1,394 @@
+
+### Database backups
+
+Databases have a backup schedule. Databases are not considered backed up unless the backups are tested regularly. There is a process documented for restore.
+
+### 4 eyes principle
+
+Pull Requests are only merged when at least one other person has reviewed and agreed that the PR is ok.
+
+### Basic health endpoint
+
+All services have a basic `health` endpoint returning back a `200OK` when the service is running.
+
+### Refinement
+
+It is important that stories are refined and that those at the top of the list are in a state where they can be picked up. This skill asserts that the team has a process and techniques that make sure that this is always the case. Typically, this is only achieved if stories are worked on collaboratively with people with different skill and knowledge sets.  
+
+Some recommended techniques for knowledge mining:
+
+- 3 Amigos: Divide and conquer so the team can refine stories in smaller groups. This is often more effective and so any final refinement is done with stories fleshed out from multiple points of view.
+- Example mapping: A great way to flesh out stories and give clear examples. Great for identifying edge-cases and expected behaviour. Also easy to do collaboratively.
+- EventStorming: A great technique for knowledge mining. It makes clear current process to all involved and makes it easy to talk about needed changes or dive into new behaviour.
+
+### Infrastructure as Code
+
+All infrastructure should be created via selected code provider such as Terraform.
+
+### Validated configuration
+
+Configuration should be validated, at latest, at application startup. If you use conventions, it is useful to place the environment in the names so they can be validated for the current environment. For this reason it is best to have them at the start or end of the name.
+### Resources are tagged
+
+Resources should be tagged with a convention followed for all resources of each type. These should be enforced by policy automation.
+
+### Smoke tests
+
+A section of tests should be marked as smoke tests that run against an environment. These tests represent such fundamental functionality of the system that if any of those tests fail, the deployed version is considered invalid and rolled back. One consideration is configuration issues, so automated checking as much configuration before deployment is valuable.
+
+### Dashboards I
+
+A central dashboard with all owned applications and the current status and uptime SLO of that application.
+
+### Contract Testing
+
+The contracts between services should be tested and checked for breaking changes. There are some advanced techniques for this but this adds complications that tend to move them downstream on the software delivery pipeline. A simple and early test is to serialize the expected response and add a unit test that checks for backward compatibility (no rename or removal of properties. No type changes of properties.). 
+
+### API client keys
```

Comment:
Is API client keys a solution, for the authorisation problem? How about other mechanisms?